packages/types/src/shared.ts

```diff
@@ -1024,7 +1024,7 @@ export type Localization = Partial<Record<Locales, string>>
 
 export type AtLeastOne<T, U = { [K in keyof T]: Pick<T, K> }> = Partial<T> & U[keyof U]
 export type CamelCase<S extends string> = S extends `${infer T}_${infer U}` ? `${T}${Capitalize<CamelCase<U>>}` : S
-export type SnakeCase<S extends string> = S extends `${infer T}${infer U}` ? `${T extends Capitalize<T> ? '_' : ''}${Lowercase<T>}${SnakeCase<U>}` : S
+export type SnakeCase<S extends string> = S extends `${infer T}${infer U}` ? `${T extends Lowercase<T> ? '' : '_'}${Lowercase<T>}${SnakeCase<U>}` : S
```

Comment:
Unrelated change?

Reply:
Looks like one. It is not. It was breaking stuff bc discord had the great of using numbers in JSON keys or something (there is a discord message I asked help on this: https://discord.com/channels/785384884197392384/785384884197392387/1219773105493377104)